docs/csharp/language-reference/tokens/interpolated.md

```diff
@@ -86,6 +86,25 @@ If an interpolated string has the type `string`, it's typically transformed into
 
 If an interpolated string has the type <xref:System.IFormattable> or <xref:System.FormattableString>, the compiler generates a call to the <xref:System.Runtime.CompilerServices.FormattableStringFactory.Create%2A?displayProperty=nameWithType> method.
 
+> [!NOTE]
+> String interpolation can cause [boxing](../../programming-guide/types/boxing-and-unboxing.md), as it happens in the following example:
+>
+> ```csharp-interactive
+> double pi = Math.PI;
+> string line = $"The pi constant is {pi:F5}";
+> Console.WriteLine(line);  // output: The pi constant is 3.14159
+> ```
+>
+> To avoid heap allocations caused by boxing, consider formatting an instance of a value type explicitly, as the following example shows:
+>
+> ```csharp-interactive
+> double pi = Math.PI;
+> string line = $"The pi constant is {pi.ToString("F5")}";
```

Comment:
On .NET Core, this latter example is actually going to allocate more, not less, than the boxing version. String interpolation like this uses String.Format, and for primitive types like double, we can format them directly into the rented buffer we use as part of the String.Format.  So beyond the overall string that's being allocated, the first example allocates a boxed double, whereas the latter example allocates a string for a formatted double, and the latter is actually larger.  On 64-bit, the former allocates 104B and the latter allocates 120B.

If someone's interested in performance around interpolation, there are lots of subtleties to understand; I worry a note like this one will actually muddy the waters more than it helps.

My $.02.

Reply:
@stephentoub indeed, @svick presented the performance difference in the [comment to the related issue](https://github.com/dotnet/docs/issues/16618#issuecomment-573315069) to show that that difference depends on .NET implementation.

> If someone's interested in performance around interpolation, there are lots of subtleties to understand; I worry a note like this one will actually muddy the waters more than it helps.

@stephentoub @BillWagner what about using that note only for its last sentence? That is, mention that both variants (boxing/no boxing) would allocate and the actual difference in performance depends on .NET implementation. Thus, do benchmarking if you're interested in performance.

If we leave the note in, I would like to make sure that it doesn't favor one of the ways.

Also, I've updated the text of the last paragraph to mention that actual difference in performance is dependent on .NET implementation.

Reply:
Also, maybe it's good to show how difficult it is. Otherwise, the "no boxing always better" approach might be non-critically applied to the string interpolation usage.

Reply:
My $.02:
.NET Core is the future, especially when someone is focused on this kind of perf difference.  I would suggest this PR just be closed. It's great to tell people to benchmark when they're looking at performance, but that's not specific to string interpolation.

Reply:
@stephentoub thank you for clarifying comments
@BillWagner I'm fine with both closing (and closing related issue then) and merging this PR